docroot/modules/custom/foia_webform/src/FoiaSubmissionQueueingService.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Drupal\foia_webform;
+
+use Drupal\foia_request\Entity\FoiaRequestInterface;
+use Drupal\Core\Queue\QueueFactory;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Class FoiaSubmissionQueueingService.
+ */
+class FoiaSubmissionQueueingService implements FoiaSubmissionQueueingServiceInterface {
+
+  /**
+   * The core Queue factory.
+   *
+   * @var \Drupal\Core\Queue\QueueFactory
+   */
+  protected $queueFactory;
+
+  /**
+   * A logger instance.
+   *
+   * @var \Psr\Log\LoggerInterface
+   */
+  protected $logger;
+
+  /**
+   * Constructs the Queueing service.
+   *
+   * @param \Drupal\Core\Queue\QueueFactory $queueFactory
+   *   The core Queue Factory.
+   * @param \Psr\Log\LoggerInterface $logger
+   *   The core Logger interface.
+   */
+  public function __construct(QueueFactory $queueFactory, LoggerInterface $logger) {
+    $this->queueFactory = $queueFactory;
+    $this->logger = $logger;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function enqueue(FoiaRequestInterface $foiaRequest) {
+    // @var QueueInterface $queue
```

Comment:
I think the right syntax for type hinting is:
```
/** @var \Drupal\Core\Queue\QueueInterface $foiaSubmissionsQueue */
```
With that being said, I don't think we actually need the type hint, since the `$queueFactory` explicitly returns an object of type `QueueInterface`.